imports/api/employee.js

```diff
@@ -1,60 +1,312 @@
 import { Meteor } from "meteor/meteor";
-import { Employees } from "./db";
+import { Employees, Tasks } from "./db";
 import EMPLOYEESAPI from "../constant/methods/employeesAPI";
+import ROLE from "../constant/role";
+import { isAuthenticated, getId, isAdmin } from "../util/authUtil";
+import { AuthError, EmployeeError, TaskError } from "../constant/error";
+import { EmployeeMessage } from "../constant/message";
+import { removeElement, addToList } from "../util/arrayUtil";
 
-export const insertEmployee = (db, employee) => {
-	//admin only action, check auth status
-	console.log("inserting employee: ", employee);
+if (Meteor.isServer) {
+	Meteor.publish("employees", () => Employees.find());
+}
+
+export const insertEmployee = (
+	db,
+	accountId,
+	firstName,
+	lastName,
+	role,
+) => {
+	if (!isAuthenticated()) {
+		throw new Meteor.Error(AuthError.NOT_AUTH);
+	}
+	if (!isAdmin(Employees) || !accountId) {
```

Comment:
I handle the case where `getId` in `authUtil` does not return anything here. However, seems like I keep getting this error by running unit test. Not too sure if I didn't implement `getId` correctly or didn't handle the `empty id` case correctly. @nerocui